test/controllers/api/v2/activation_keys_controller_test.rb

```diff
@@ -90,18 +91,125 @@ def test_create_protected
+      }
+
+      assert_response :success
+      assert_template 'katello/api/v2/common/create'
+      response = JSON.parse(@response.body)
+      assert response.key?('unlimited_hosts')
+      assert response.key?('max_hosts')
+      refute response['unlimited_hosts']
+      assert_equal max_hosts, response['max_hosts']
+    end
+
+    test_attributes :pid => '749e0d28-640e-41e5-89d6-b92411ce73a3'
+    def test_create_with_name
+      key_name = 'key_with_name'
+      ActivationKey.any_instance.expects(:reload)
+      assert_sync_task(::Actions::Katello::ActivationKey::Create) do |activation_key|
+        assert_equal key_name, activation_key.name
+        assert_valid activation_key
+      end
+
+      post :create, params: { :organization_id => @organization.id, :activation_key => {:name => key_name} }
+
+      assert_response :success
+      assert_template 'katello/api/v2/common/create'
+      response = JSON.parse(@response.body)
+      assert response.key?('name')
+      assert_equal key_name, response['name']
+    end
+
+    test_attributes :pid => '64d93726-6f96-4a2e-ab29-eb5bfa2ff8ff'
+    def test_create_with_description
+      key_description = 'key_with_description'
+      ActivationKey.any_instance.expects(:reload)
+      assert_sync_task(::Actions::Katello::ActivationKey::Create) do |activation_key|
+        assert_equal key_description, activation_key.description
+        assert_valid activation_key
+      end
+
+      post :create, params: {
+        :organization_id => @organization.id,
+        :activation_key => {:name => 'new key', :description => key_description}
+      }
+
+      assert_response :success
+      assert_template 'katello/api/v2/common/create'
+      response = JSON.parse(@response.body)
+      assert response.key?('description')
+      assert_equal key_description, response['description']
+    end
+
+    test_attributes :pid => 'a9e756e1-886d-4f0d-b685-36ce4247517d'
+    def test_should_not_create_with_no_hosts_limit
+      post :create, params: {
+        :organization_id => @organization.id,
+        :activation_key => {:name => 'limited Key', :unlimited_hosts => false}
+      }
+      assert_response :unprocessable_entity
+      assert_match 'Validation failed: Max hosts cannot be nil', @response.body
+    end
+
+    test_attributes :pid => 'c018b177-2074-4f1a-a7e0-9f38d6c9a1a6'
+    def test_should_not_create_with_invalid_hosts_limit
+      post :create, params: {
+        :organization_id => @organization.id,
+        :activation_key => {:name => 'limited Key', :unlimited_hosts => false, :max_hosts => 0}
+      }
+      assert_response :unprocessable_entity
+      assert_match 'Validation failed: Max hosts cannot be less than one', @response.body
+    end
+
+    test_attributes :pid => '71b9b000-b978-4a95-b6f8-83c09ed39c01'
+    def test_should_not_create_unlimited_and_invalid_max_hosts
+      # BZ: 1156555
```

Comment:
mind removing the bz reference? (or changing it to an upstream issue)

Reply:
@jlsherrill  removed the BZ info and rebased